### PR TITLE
output: update generated resource files

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds tasks.py and permissions.py.
* Updates README.rst.
* Fix whitespaces and blank lines.
* Updates license into some files.
* Adds search class and methods into api.py.
* Removes fetchers, minters, providers.

Co-Authored-by: Laurent Dubois <laurent.dubois@itld-solutions.be>

### CODE REVIEW

File: {{ cookiecutter.resource_name }}/api.py [hunk 1]
```diff
@@ -15,20 +15,54 @@
...
+from ..fetchers import id_fetcher
+from ..minters import id_minter
+from ..providers import Provider
+from ..api import IlsRecord, IlsRecordsSearch
+
+# provider
+{{ cookiecutter.class_name }}Provider = type(
+    '{{ cookiecutter.class_name }}Provider',
+    (Provider,),
+    dict(identifier={{ cookiecutter.class_name }}Identifier, pid_type='pers')
```
Comment (BadrAly) [inline {{ cookiecutter.resource_name }}/api.py]: `pid_type='pers')` This you need to update pid type  here

File: {{ cookiecutter.resource_name }}/api.py [hunk 2]
```diff
@@ -15,20 +15,54 @@
...
 
 class {{ cookiecutter.class_name }}(IlsRecord):
     """{{ cookiecutter.class_name }} class."""
 
     minter = {{ cookiecutter.name }}_id_minter
     fetcher = {{ cookiecutter.name }}_id_fetcher
     provider = {{ cookiecutter.class_name }}Provider
+
+    @classmethod
+    def create(cls, data, id_=None, delete_pid=False,
```
Comment (BadrAly) [inline {{ cookiecutter.resource_name }}/api.py:56]: We can add in the `README` file a note that if `create` and `update` functions are similar to `rero_ils.modules.api` then , these two functions can be safely removed

**Review by BadrAly — CHANGES_REQUESTED** [review]

File: {{ cookiecutter.resource_name }}/api.py (hunk 1 above)
Comment (lauren-d) [reply {{ cookiecutter.resource_name }}/api.py]: Done

File: {{ cookiecutter.resource_name }}/api.py (hunk 2 above)
Comment (lauren-d) [reply {{ cookiecutter.resource_name }}/api.py:56]: Done

**Review by BadrAly — APPROVED** [review]

**Pull request merged**